server/src/test/java/org/elasticsearch/transport/OutboundHandlerTests.java

```diff
@@ -317,6 +322,101 @@ public void onResponseSent(long requestId, String action, Exception error) {
         assertEquals("header_value", header.getHeaders().v1().get("header"));
     }
 
+    public void testFailToSendResponse() {
+        TransportVersion version = TransportVersionUtils.randomVersion();
+        String action = randomAlphaOfLength(10);
+        long requestId = randomLongBetween(0, 300);
+        var response = new ReleasbleTestResponse(randomAlphaOfLength(10)) {
+            @Override
+            public void writeTo(StreamOutput out) {
+                throw new CircuitBreakingException("simulated", CircuitBreaker.Durability.TRANSIENT);
+            }
+        };
+
+        AtomicLong requestIdRef = new AtomicLong();
+        AtomicReference<String> actionRef = new AtomicReference<>();
+        AtomicReference<TransportResponse> responseRef = new AtomicReference<>();
+        AtomicReference<Exception> exceptionRef = new AtomicReference<>();
+        handler.setMessageListener(new TransportMessageListener() {
+            @Override
+            public void onResponseSent(long requestId, String action, TransportResponse response) {
+                assertThat(requestIdRef.get(), equalTo(0L));
+                requestIdRef.set(requestId);
+                assertNull(actionRef.get());
+                actionRef.set(action);
+                assertNull(responseRef.get());
+                responseRef.set(response);
+            }
+
+            @Override
+            public void onResponseSent(long requestId, String action, Exception error) {
```

Comment:
Could we also assert that `channel.getMessageCaptor()` has seen an exception message here?